src/compiler/utilities.ts

```diff
@@ -6197,6 +6197,58 @@ namespace ts {
         return scriptKind === ScriptKind.TSX || scriptKind === ScriptKind.JSX || scriptKind === ScriptKind.JS || scriptKind === ScriptKind.JSON ? LanguageVariant.JSX : LanguageVariant.Standard;
     }
 
+    /**
+     * This is a somewhat unavoidable full tree walk to locate a JSX tag - `import.meta` requires the same,
+     * but we avoid that walk (or parts of it) if at all possible using the `PossiblyContainsImportMeta` node flag.
+     * Unfortunately, there's no `NodeFlag` space to do the same for JSX.
+     */
+    function walkTreeForJSXTags(node: Node): Node | undefined {

```

Comment:
Because it's used by `getSetExternalModuleIndicator` which is in `utilities`, which is in `utilities` because it's used _everywhere_ (well, in program and a bunch of services endpoints anyway).